docs/src/howto/async-commit-merge.md

```diff
@@ -0,0 +1,436 @@
+    }
+  }'
+```
+
+**Response (202 Accepted):**
+```json
+{
+  "id": "op_a1b2c3d4e5f6"
+}
+```
+
+!!! important "Save the operation ID"
+Store the `id` for status queries. The ID is required to retrieve results.
+
+### Checking Commit Status
+
+**Endpoint:** `GET /repositories/{repository}/branches/{branch}/commits/async/{operation_id}/status`
+
+**Request:**
+```bash
+curl "https://lakefs.example.com/api/v1/repositories/example-repo/branches/main/commits/async/op_a1b2c3d4e5f6/status" \
+  -H "Authorization: Bearer ${LAKEFS_ACCESS_TOKEN}"
+```
+
+**Response (In Progress):**
+```json
+{
+  "task_id": "op_a1b2c3d4e5f6",
+  "completed": false,
+  "update_time": "2024-12-19T10:02:15Z"
+}
+```
+
+**Response (Completed Successfully):**
+```json
+{
+  "task_id": "op_a1b2c3d4e5f6",
+  "completed": true,
+  "update_time": "2024-12-19T10:05:43Z",
+  "result": {
+    "id": "c1a2b3c4d5e6f7g8h9i0",
+    "parents": ["b1a2b3c4d5e6f7g8h9i0"],
+    "committer": "user@example.com",
+    "message": "Daily data ingestion",
+    "creation_date": 1734604543,
+    "meta_range_id": "abc123def456",
+    "metadata": {
+      "pipeline": "etl-prod",
+      "run_id": "2024-12-19"
+    },
+    "generation": 42,
+    "version": 1
+  }
+}
+```
+
+**Response (Failure due to branch protection):**
+```json
+{
+  "task_id": "op_a1b2c3d4e5f6",
+  "completed": true,
+  "update_time": "2024-12-19T10:04:22Z",
+  "error": {
+    "message": "cannot commit to protected branch"
+  },
+  "status_code": 403
+}
+```
+
+### Submitting an Async Merge
+
+**Endpoint:** `POST /repositories/{repository}/refs/{sourceRef}/merge/{destinationBranch}/async`
+
+**Request:**
+```bash
+curl -X POST "https://lakefs.example.com/api/v1/repositories/example-repo/refs/feature-branch/merge/main/async" \
+  -H "Authorization: Bearer ${LAKEFS_ACCESS_TOKEN}" \
+  -H "Content-Type: application/json" \
+  -d '{
+    "message": "Merge feature branch into main",
+    "metadata": {
+      "approver": "tech-lead",
+      "ticket": "PROJ-123"
+    }
+  }'
+```
+
+**Response (202 Accepted):**
+```json
+{
+  "id": "op_z9y8x7w6v5u4"
+}
+```
+
+### Checking Merge Status
+
+**Endpoint:** `GET /repositories/{repository}/refs/{sourceRef}/merge/{destinationBranch}/async/{operation_id}/status`
+
+**Request:**
+```bash
+curl "https://lakefs.example.com/api/v1/repositories/example-repo/refs/feature-branch/merge/main/async/op_z9y8x7w6v5u4/status" \
+  -H "Authorization: Bearer ${LAKEFS_ACCESS_TOKEN}"
+```
+
+**Response (Completed Successfully):**
+```json
+{
+  "task_id": "op_z9y8x7w6v5u4",
+  "completed": true,
+  "update_time": "2024-12-19T14:35:12Z",
+  "result": {
+    "reference": "c5d6e7f8g9h0i1j2k3l4"
+  }
+}
+```
+
+**Response (Failure due to merge conflict):**
+```json
+{
+  "task_id": "op_z9y8x7w6v5u4",
+  "completed": true,
+  "update_time": "2024-12-19T14:32:45Z",
+  "error": {
+    "message": "Merge conflict detected between source and destination branches"
+  },
+  "status_code": 409
+}
+```
+
+## How It Works
```

Comment:
I really think that this section should be placed _before_ the `Using lakectl` part - 
It's a good overview that explains how the feature works, before diving into the API details.